core/src/main/java/com/scalar/db/api/DistributedTransactionManager.java

```diff
@@ -56,9 +57,11 @@ public interface DistributedTransactionManager {
    * Begins a new transaction.
    *
    * @return {@link DistributedTransaction}
+   * @throws RetriableTransactionException if something retriable situation happens. You can retry
+   *     the same transaction from the beginning in this case
    * @throws TransactionException if beginning the transaction fails
    */
-  DistributedTransaction begin() throws TransactionException;
+  DistributedTransaction begin() throws RetriableTransactionException, TransactionException;
```

Comment:
Made this method throw `RetriableTransactionException`. Since `RetriableTransactionException` is a sub-class of `TransactionException`, it's not an incompatible changes.